Unity Events/CollisionEventManager.cs

```diff
@@ -0,0 +1,172 @@
+        {
+            collision2DEvent.HandleEvent(collision2DEvent.onTriggerEnter, other.gameObject);
+        }
+    }
+
+    private void OnTriggerStay2D(Collider2D other)
+    {
+        foreach (CollisionEvent collision2DEvent in events)
+        {
+            collision2DEvent.HandleEvent(collision2DEvent.onTriggerStay, other.gameObject);
+        }
+    }
+
+    private void OnTriggerExit2D(Collider2D other)
+    {
+        foreach (CollisionEvent collision2DEvent in events)
+        {
+            collision2DEvent.HandleEvent(collision2DEvent.onTriggerExit, other.gameObject);
+        }
+    }
+
+    private void OnCollisionEnter2D(Collision2D other)
+    {
+        foreach (CollisionEvent collision2DEvent in events)
+        {
+            collision2DEvent.HandleEvent(collision2DEvent.onCollisionEnter, other.gameObject);
+        }
+    }
+    private void OnCollisionStay2D(Collision2D other)
+    {
+        foreach (CollisionEvent collision2DEvent in events)
+        {
+            collision2DEvent.HandleEvent(collision2DEvent.onCollisionStay, other.gameObject);
+        }
+    }
+
+    private void OnCollisionExit2D(Collision2D other)
+    {
+        foreach (CollisionEvent collision2DEvent in events)
+        {
+            collision2DEvent.HandleEvent(collision2DEvent.onCollisionExit, other.gameObject);
+        }
+    }
+
+    #endregion
+}
+
+#if UNITY_EDITOR
+[CustomEditor(typeof(CollisionEventManager), true)]
+[CanEditMultipleObjects]
+public class CollisionEventManagerEditor : Editor
+{
+    private CollisionEventManager manager;
+    private List<Editor> cachedEditors = new List<Editor>();
+    private List<bool> showEvents = new List<bool>();
+    private UnityEvent garbageCollector = new UnityEvent();
+
+    public override void OnInspectorGUI()
+    {
+        if (manager == null) manager = (CollisionEventManager)target;
+        GUI.color = Color.green;
+        if (GUILayout.Button("Add event"))
+            manager.events.Add(CreateInstance<CollisionEvent>());
+        GUI.color = Color.white;
+
+        for (int i = 0; i < manager.events.Count; i++)
+        {
+            if (cachedEditors.Count < manager.events.Count) cachedEditors.Add(new Editor());
+            if (showEvents.Count < manager.events.Count) showEvents.Add(false);
+            showEvents[i] = EditorGUILayout.Foldout(showEvents[i], "Event " + (i + 1));
+            if (showEvents[i])
+            {
+                GUI.color = Color.red;
+                if (GUILayout.Button("Delete")) RemoveEvent(manager.events[i]);
+                GUI.color = Color.white;
+                Editor tempEditor = cachedEditors[i];
+                CreateCachedEditor(manager.events[i], typeof(EventTriggerEditor), ref tempEditor);
+                if (tempEditor != null) tempEditor.OnInspectorGUI();
+                cachedEditors[i] = tempEditor;
+            }
+        }
+
+        //Garbage collector? overkill? yes. Unnecessary? maybe. Better alternatives? Definitely.
```

Comment:
Stuff like this for example is just very unnecessary and you might want to adjust even though it "works"